crates/matrix-sdk-ui/tests/integration/timeline/profiles.rs

```diff
@@ -0,0 +1,200 @@
+
+use std::{sync::Arc, time::Duration};
+
+use assert_matches::assert_matches;
+use matrix_sdk::config::SyncSettings;
+use matrix_sdk_test::{
+    async_test, EventBuilder, JoinedRoomBuilder, SyncResponseBuilder, ALICE, BOB, CAROL,
+    DEFAULT_TEST_ROOM_ID,
+};
+use matrix_sdk_ui::timeline::{RoomExt, TimelineDetails};
+use ruma::events::room::{
+    member::{MembershipState, RoomMemberEventContent},
+    message::RoomMessageEventContent,
+};
+use serde_json::json;
+use wiremock::{
+    matchers::{method, path_regex},
+    Mock, ResponseTemplate,
+};
+
+use crate::{logged_in_client, mock_sync};
+
+#[async_test]
+async fn update_sender_profiles() {
+    let (client, server) = logged_in_client().await;
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+
+    let event_builder = EventBuilder::new();
+    let mut sync_builder = SyncResponseBuilder::new();
+    sync_builder.add_joined_room(JoinedRoomBuilder::new(&DEFAULT_TEST_ROOM_ID));
+
+    mock_sync(&server, sync_builder.build_json_sync_response(), None).await;
+    let _response = client.sync_once(sync_settings.clone()).await.unwrap();
+    server.reset().await;
+
+    let room = client.get_room(&DEFAULT_TEST_ROOM_ID).unwrap();
+    let timeline = Arc::new(room.timeline().await);
+
+    sync_builder.add_joined_room(
+        JoinedRoomBuilder::new(&DEFAULT_TEST_ROOM_ID)
+            // Alice accepts an invite into the room
+            .add_timeline_event(event_builder.make_sync_state_event(
+                &ALICE,
+                ALICE.as_str(),
+                RoomMemberEventContent::new(MembershipState::Join),
+                Some(RoomMemberEventContent::new(MembershipState::Invite)),
+            ))
+            // Bob sends a text message
+            .add_timeline_event(event_builder.make_sync_message_event(
+                &BOB,
+                RoomMessageEventContent::text_plain("text message event"),
+            ))
+            // Carol kicks bob
+            .add_timeline_event(event_builder.make_sync_state_event(
+                &CAROL,
+                BOB.as_str(),
+                RoomMemberEventContent::new(MembershipState::Leave),
+                Some(RoomMemberEventContent::new(MembershipState::Join)),
+            )),
+    );
+
+    mock_sync(&server, sync_builder.build_json_sync_response(), None).await;
+    let _response = client.sync_once(sync_settings.clone()).await.unwrap();
+
+    let timeline_items = timeline.items().await;
+    assert_eq!(timeline_items.len(), 4);
+    assert!(timeline_items[0].is_day_divider());
+    // We have seen a member event for alice from alice => we have a profile
+    // albeit empty.
+    assert_matches!(
+        timeline_items[1].as_event().unwrap().sender_profile(),
+        TimelineDetails::Ready(_)
+    );
+    // We haven't seen a member event for bob from himself.
+    assert_matches!(
+        timeline_items[2].as_event().unwrap().sender_profile(),
+        TimelineDetails::Ready(_) // ?
```

Comment:
As discussed in chat, this is the current expected behavior. We'll use the state event if we don't have anything better.